core/templates/pages/exploration-player-page/services/stats-reporting.service.spec.ts

```diff
@@ -60,20 +60,20 @@ describe('Stats reporting service ', () => {
   });
 
   beforeEach(() => {
-    spyOn(messengerService, 'sendMessage').and.returnValue(null);
-    spyOn(siteAnalyticsService, 'registerNewCard').and.returnValue(null);
+    spyOn(messengerService, 'sendMessage').and.callThrough();
+    spyOn(siteAnalyticsService, 'registerNewCard').and.callThrough();
     spyOn(siteAnalyticsService, 'registerStartExploration')
-      .and.returnValue(null);
+      .and.returnValue();
     spyOn(siteAnalyticsService, 'registerFinishExploration')
-      .and.returnValue(null);
+      .and.returnValue();
     spyOn(siteAnalyticsService, 'registerCuratedLessonCompleted')
-      .and.returnValue(null);
+      .and.returnValue();
     spyOn(playthroughService, 'recordExplorationStartAction')
-      .and.returnValue(null);
+      .and.returnValue();
     spyOn(playthroughService, 'recordExplorationQuitAction')
-      .and.returnValue(null);
+      .and.returnValue();
     spyOn(playthroughService, 'storePlaythrough')
-      .and.returnValue(null);
+      .and.returnValue();
```

Comment:
replace `.and.returnValue()` with `.and.callThrough()`.

Reply:
Done change in whole codebase.